internal/search/query/parser.go

```diff
@@ -825,12 +825,20 @@ func ParseAndOr(in string) ([]Node, error) {
 	return newOperator(nodes, And), nil
 }
 
+type ParserOptions struct {
+	SearchType SearchType
+
+	// defines whether field values of supported
+	// fields are treated as regex or glob.
```

Comment:
```suggestion
	// treat repo, file, or repohasfile values as glob syntax if true.
```

Not attached to the particular wording, just feels like it could be shorter.

Reply:
your description is much clearer